apps/meteor/tests/e2e/account-forgetSessionOnWindowClose.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { DEFAULT_USER_CREDENTIALS } from './config/constants';
+import { Registration } from './page-objects';
+import { test, expect } from './utils/test';
+
+test.describe.serial('Forget session on window close setting', () => {
+	let poRegistration: Registration;
+
+	test.beforeEach(async ({ page }) => {
+		poRegistration = new Registration(page);
+
+		await page.goto('/home');
+	});
+
+	test.describe('Setting off', async () => {
+		test.beforeAll(async ({ api }) => {
+			await api.post('/settings/Accounts_ForgetUserSessionOnWindowClose', { value: false });
+		});
+
+		test('Login using credentials and reload to stay logged in', async ({ page }) => {
+			await poRegistration.username.type('user1');
+			await poRegistration.inputPassword.type(DEFAULT_USER_CREDENTIALS.password);
+			await poRegistration.btnLogin.click();
+
+			await expect(page.locator('role=heading[name="Welcome to Rocket.Chat"]')).toBeVisible();
+
+			await page.reload();
+
+			await expect(page.locator('role=heading[name="Welcome to Rocket.Chat"]')).toBeVisible();
+		});
+	});
+
+	test.describe('Setting on', async () => {
+		test.beforeAll(async ({ api }) => {
+			await api.post('/settings/Accounts_ForgetUserSessionOnWindowClose', { value: true });
+		});
+
+		test.afterAll(async ({ api }) => {
+			await api.post('/settings/Accounts_ForgetUserSessionOnWindowClose', { value: false });
+		});
+
+		test('Login using credentials and reload to get logged out', async ({ page }) => {
+			await poRegistration.username.type('user1');
+			await poRegistration.inputPassword.type(DEFAULT_USER_CREDENTIALS.password);
+			await poRegistration.btnLogin.click();
+
+			await expect(page.locator('role=heading[name="Welcome to Rocket.Chat"]')).toBeVisible();
+
+			await page.reload();
+
+			await expect(page.locator('role=button[name="Login"]')).toBeVisible();
```

Comment:
I noticed that after reloading the page the user only stays logged out for a few seconds before automatically logging back in. Is this the expected behavior? The current test stops as soon as the login page appears, but it seems like the test would fail if it waited a few more seconds

Reply:
You are right, session storage is valid across page reloads.

Pushed new tests, to open a new tab and check the status, since session storage doesn't work across tabs.